Example/Storage/Tests/Integration/FIRStorageIntegrationTests.m

```diff
@@ -454,6 +454,86 @@ - (void)testCancelDownload {
     [self waitForExpectations];
 }
 
+- (void)testUpdateMetadata {
+  XCTestExpectation *expectation =
+    [self expectationWithDescription:@"testUpdateMetadata"];
+
+  FIRStorageReference *ref = [self.storage referenceWithPath:@"ios/public/1mb"];
+
+  // Update all available metadata
+  FIRStorageMetadata *metadata = [[FIRStorageMetadata alloc] init];
+  metadata.cacheControl = @"cache-control";
+  metadata.contentDisposition = @"content-disposition";
+  metadata.contentEncoding = @"gzip";
+  metadata.contentLanguage = @"de";
+  metadata.contentType = @"content-type-a";
+  metadata.customMetadata = @{ @"a" : @"b" };
+
+  void (^assertMetadata)(FIRStorageMetadata*, NSString*, NSDictionary*) =
+      ^(FIRStorageMetadata* actualMetadata, NSString* expectedContentType, NSDictionary* expectedCustomMetadata) {
+    XCTAssertEqualObjects(actualMetadata.cacheControl, @"cache-control");
+    XCTAssertEqualObjects(actualMetadata.contentDisposition, @"content-disposition");
+    XCTAssertEqualObjects(actualMetadata.contentEncoding, @"gzip");
+    XCTAssertEqualObjects(actualMetadata.contentLanguage, @"de");
+    XCTAssertEqualObjects(actualMetadata.contentType, expectedContentType);
+    for(id key in expectedCustomMetadata) {
+      XCTAssertEqualObjects([actualMetadata.customMetadata objectForKey:key], [expectedCustomMetadata objectForKey:key]);
+    }
+  };
+
+  [ref updateMetadata:metadata completion:^ (FIRStorageMetadata* updatedMetadata, NSError* error ){
```

Comment:
Also, is this worth breaking into multiple tests? This test is massive, and seems to be testing multiple things